oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -481,6 +482,14 @@ static ExternalAccountCredentials fromJson(
         .build();
   }
 
+  private static <T> T getOptional(Map<String, Object> json, String fieldName, Class<T> clazz) {
+    Object value = json.get(fieldName);
+    if (value == null || Data.isNull(value)) {
+      return null;
+    }
+    return clazz.cast(value);
```

Comment:
Can we just cast this as a `(String)` when we return? Or is there a specific reason to use `Class.cast()`?

Reply:
It's also handling impersonationOptionsMap, which is a Map, not a String.